Dockerfile

```
@@ -1,24 +1,36 @@
 # Use a recent Node.js LTS version
+FROM node:22-alpine AS builder
+
+WORKDIR /app
+
+# Install yt-dlp and native build tools (e.g. sodium)
+RUN apk add --no-cache python3 py3-pip ffmpeg build-base autoconf automake libtool g++ \
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)